Error: browserType.launch: Chromium distribution 'chrome' is not found on debian bullseye image

Hi,
I have a Dockerfile which builds a Debian bullseye image for a ruby on rails application,
I have a GitHub actions script that pulls the image from docker hub and installs playwright to run the tests in a container environment,
I have trouble installing Chrome and I keep getting the following error:  Error: browserType.launch: Chromium distribution 'chrome' is not found at /opt/google/chrome/chrome
    Run "npx playwright install chrome"

From the Playwright documentation it's written that Playwright now supports Debian 11 Bullseye on x86_64 for Chromium, Firefox and WebKit,

it fails when trying to run the tests because it says that chrome is not installed and that I need to run npx install chrome, when I do run npx playwright install chrome it returns an error that ERROR: cannot install on debian distribution - only Ubuntu is supported,
although I am using a base image that runs on debian 11 so I'm a bit confused, what exactly am I missing?


error1:
Run npx playwright install chrome
++ arch
+ [[ x86_64 == \a\a\r\c\h\6\4 ]]
+ [[ ! -f /etc/os-release ]]
++ bash -c 'source /etc/os-release && echo $ID'
+ ID=debian
ERROR: cannot install on debian distribution - only Ubuntu is supported
+ [[ debian != \u\b\u\n\t\u ]]
+ echo 'ERROR: cannot install on debian distribution - only Ubuntu is supported'
+ exit 1
Failed to install browsers
Error: Failed to install chrome
Error: Process completed with exit code 1.

error2:

    Error: browserType.launch: Chromium distribution 'chrome' is not found at /opt/google/chrome/chrome
    Run "npx playwright install chrome"



here is my github actions workflow:
name: Playwright Tests
on:
  push:
    branches:
      - "*"
  pull_request:
    types:
      - closed
    branches: [dev, master]
env:
  IMAGE_TAG: ${{ github.sha }}
  PLAYWRIGHT_DIRECTORY: playwright
  DOCKER_HUB_USER: ****
  DOCKER_HUB_PASSWORD:****
jobs:
  test_build:
    if: github.event_name == 'push'
    timeout-minutes: 60
    runs-on: ubuntu-20.04
    container:
      image:almond-rails-app:bull
      credentials:
        username: ${{ env.DOCKER_HUB_USER }}
        password: ${{ env.DOCKER_HUB_PASSWORD }}
    strategy:
      matrix:
        modules: ["@SANITY"]
    steps:
      - uses: actions/checkout@v3
      - name: run the container via docker compose
        run: |
          docker compose -f docker-compose.yml up -d
          echo "Server is Up and running"
      - name: Install playwright dependencies and run tests
        run: npm ci
      - name: install playwright chromium browser
        run: npx playwright install chrome
      - name: run playwright tests
        run: npx playwright test --grep ${{ matrix.modules }}
      - uses: actions/upload-artifact@v3
        if: always()
        with:
          path: uiAutomation/Playwright/playwright-report/
          retention-days: 30
      - name: login to dockerhub
        uses: docker/login-action@v2
        with:
          username: ${{ env.DOCKER_HUB_USER }}
          password: ${{ env.DOCKER_HUB_PASSWORD }}
      - name: build docker image
        run: |
          echo "building the image"
          docker build -f Dockerfile -t almond-rails-app:${{ env.IMAGE_TAG }} .
          echo "pushing to dockerhub"
          docker push almond-rails-app:${{ env.IMAGE_TAG }}
        if: github.event.pull_request.merged


my dockerfile

FROM ruby:3.2.2-slim-bullseye as builder

RUN apt-get update && \
    apt-get install -y \
    build-essential \
    npm \
    git \
    libpq-dev \
    libvips42 \
    pkg-config \
    lsb-release \
    imagemagick \ 
    libmagickwand-dev \
    curl

WORKDIR /app
RUN curl -sL https://deb.nodesource.com/setup_18.x | bash - && apt-get update && apt-get install -y nodejs \
    && apt-get upgrade -qq \
    && apt-get clean /var/lib/apt/lists/*

RUN gem update --system 
RUN gem install bundler -v 2.3.17
RUN gem install rails -v '7.0.6'
RUN gem install rmagick -v '4.2.5' --source 'https://rubygems.org/'
COPY Gemfile Gemfile.lock . / 
RUN bundle config --local without "development test"
RUN bundle install --jobs 4 --retry 3 
COPY app/assets app/assets
COPY bin bin
COPY config config
ENV RAILS_ENV=production
COPY . .

###########################################
FROM builder as final-image
WORKDIR /app
ENV RAILS_ENV=production
COPY --from=builder /usr/local/bundle/ /usr/local/bundle/
COPY . .
CMD [ "rails", "s", "-b" ,"0.0.0.0" ]


playwright config:
import { defineConfig, devices } from '@playwright/test';
require('dotenv').config()

export default defineConfig({
  testDir: './tests',
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: 'html',
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */

  use: {
    /* Base URL to use in actions like `await page.goto('/')`. */
    // baseURL: 'http://127.0.0.1:3000',

    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on',
    screenshot: 'on',
    video: 'on',
    viewport: { height: 1400, width: 900 },
    launchOptions: {
      headless: process.env.CI ? true : false,
      slowMo: 500,

    }
  },


  /* Configure projects for major browsers */
  projects: [
    {
      name: 'Google Chrome',
      use: { ...devices['Desktop Chrome'], channel: 'chrome', viewport: { width: 1400, height: 900 } }, // or 'chrome-beta'
    },

--- a/packages/playwright-core/bin/reinstall_chrome_beta_linux.sh
+++ b/packages/playwright-core/bin/reinstall_chrome_beta_linux.sh
@@ -13,8 +13,8 @@ if [[ ! -f "/etc/os-release" ]]; then
 fi
 
 ID=$(bash -c 'source /etc/os-release && echo $ID')
-if [[ "${ID}" != "ubuntu" ]]; then
-  echo "ERROR: cannot install on $ID distribution - only Ubuntu is supported"
+if [[ "${ID}" != "ubuntu" && "${ID}" != "debian" ]]; then
+  echo "ERROR: cannot install on $ID distribution - only Ubuntu and Debian are supported"
   exit 1
 fi
 

--- a/packages/playwright-core/bin/reinstall_chrome_stable_linux.sh
+++ b/packages/playwright-core/bin/reinstall_chrome_stable_linux.sh
@@ -13,8 +13,8 @@ if [[ ! -f "/etc/os-release" ]]; then
 fi
 
 ID=$(bash -c 'source /etc/os-release && echo $ID')
-if [[ "${ID}" != "ubuntu" ]]; then
-  echo "ERROR: cannot install on $ID distribution - only Ubuntu is supported"
+if [[ "${ID}" != "ubuntu" && "${ID}" != "debian" ]]; then
+  echo "ERROR: cannot install on $ID distribution - only Ubuntu and Debian are supported"
   exit 1
 fi
 

--- a/packages/playwright-core/bin/reinstall_msedge_beta_linux.sh
+++ b/packages/playwright-core/bin/reinstall_msedge_beta_linux.sh
@@ -14,8 +14,8 @@ if [[ ! -f "/etc/os-release" ]]; then
 fi
 
 ID=$(bash -c 'source /etc/os-release && echo $ID')
-if [[ "${ID}" != "ubuntu" ]]; then
-  echo "ERROR: cannot install on $ID distribution - only Ubuntu is supported"
+if [[ "${ID}" != "ubuntu" && "${ID}" != "debian" ]]; then
+  echo "ERROR: cannot install on $ID distribution - only Ubuntu and Debian are supported"
   exit 1
 fi
 

--- a/packages/playwright-core/bin/reinstall_msedge_dev_linux.sh
+++ b/packages/playwright-core/bin/reinstall_msedge_dev_linux.sh
@@ -14,8 +14,8 @@ if [[ ! -f "/etc/os-release" ]]; then
 fi
 
 ID=$(bash -c 'source /etc/os-release && echo $ID')
-if [[ "${ID}" != "ubuntu" ]]; then
-  echo "ERROR: cannot install on $ID distribution - only Ubuntu is supported"
+if [[ "${ID}" != "ubuntu" && "${ID}" != "debian" ]]; then
+  echo "ERROR: cannot install on $ID distribution - only Ubuntu and Debian are supported"
   exit 1
 fi
 

--- a/packages/playwright-core/bin/reinstall_msedge_stable_linux.sh
+++ b/packages/playwright-core/bin/reinstall_msedge_stable_linux.sh
@@ -14,8 +14,8 @@ if [[ ! -f "/etc/os-release" ]]; then
 fi
 
 ID=$(bash -c 'source /etc/os-release && echo $ID')
-if [[ "${ID}" != "ubuntu" ]]; then
-  echo "ERROR: cannot install on $ID distribution - only Ubuntu is supported"
+if [[ "${ID}" != "ubuntu" && "${ID}" != "debian" ]]; then
+  echo "ERROR: cannot install on $ID distribution - only Ubuntu and Debian are supported"
   exit 1
 fi
 

